lib/perl/Genome/Model/Tools/Sx/Trim/ByPosition.pm

```diff
@@ -0,0 +1,152 @@
+
+    Examples:
+
+    # Trim (remove) entire SEQ1
+    SEQ1
+
+    # Trim bases 1000 to 2000 from SEQ2
+    SEQ2 1000-2000
+
+    # Trim several positions from SEQ3
+    SEQ3 100-200,400-500,600-800
+
+HELP
+}
+
+sub execute {
+    my $self = shift;
+
+    my $trim_positions = $self->load_positions( $self->positions_path );
+    $self->trim_positions($trim_positions);
+
+    $self->_init;
+
+    my ($seqs, @seqs_to_write);
+    while ( $seqs = $self->_reader->read ) {
+        for my $seq ( @$seqs ) {
+            $self->trim_sequence($seq);
+            push @seqs_to_write, $seq if length($seq->{seq});
+        }
+        next if not @seqs_to_write;
+        $self->_writer->write(\@seqs_to_write);
+        @seqs_to_write = ();
+    }
+
+    return 1;
+}
+
+sub load_positions {
+    my ($class, $positions_path) = validate_pos(@_, {isa => __PACKAGE__}, {type => SCALAR});
+
+    my $fh = Genome::Sys->open_file_for_reading($positions_path);
+    my %trim_positions;
+    while ( my $line = $fh->getline ) {
+        chomp $line;
+        my ($seq_id, $positions) = split(/\s+/, $line, 2);
+        $class->fatal_message('Duplicate sequence id in trim positions! %s', $seq_id) if $trim_positions{$seq_id};
+
+        if ( not $positions ) {
+            $trim_positions{$seq_id} = [];
+            next;
+        }
+
+        for my $set ( split(/,/, $positions) ) {
+            my ($start, $stop) = split(/\-/, $set);
+            if ( $RE{num}{int}->matches($stop) ) {
+                $class->fatal_message('Invalid positions for %s! %s', $seq_id, $set) if $start == $stop or $start > $stop;
+            } elsif ( $stop ne 'end' ) {
+                $class->fatal_message('Invalid stop position for %s! %s', $seq_id, $stop);
+            }
+            push @{$trim_positions{$seq_id}}, [ $start, $stop ];
+        }
+    }
+    $fh->close;
+
+    return \%trim_positions;
+}
+
+sub keep_positions_for_sequence {
+    my ($self, $seq) = @_;
+
+    # Look up by id
+    my $trim_positions = $self->trim_positions->{ $seq->{id} };
+    if ( not $trim_positions ) {
+        # Look up by orig seq id
+        $trim_positions = $self->trim_positions->{ $seq->{orig_seq_id} } if $seq->{orig_seq_id};
+        if ( not $trim_positions ) {
+            # pcap naming
+            my $pcap_seq_id = $seq->{id};
+            $pcap_seq_id =~ s/scaffold/Contig/g;
+            $trim_positions = $self->trim_positions->{$pcap_seq_id};
+            return [ [ 0, length($seq->{seq})] ] if not $trim_positions; # keep it!
+        }
+    }
+
+    return if not @$trim_positions; # sequence indicated, but no trim positions given means trim all/keep none
+
+    my @keep_positions;
+    my $current_pos = 0;
+    for my $set ( @$trim_positions ) {
+        push @keep_positions, [ $current_pos, ( $set->[0] - $current_pos - 1 ) ];
+        $current_pos = $set->[1];
```

Comment:
Maybe this would be a good place to convert [the stop value `'end'`](https://github.com/ebelter/genome/blob/9b94526cc4f7aa12acc6ebf33950146a8423c7ab/lib/perl/Genome/Model/Tools/Sx/Trim/ByPosition.pm#L86) to an actual number? Like
```perl
$current_pos = length($seq->{seq}) if $current_pos eq 'end';
```

I guess the below condition would work anyway, but with warnings about `'end'` not being numeric.